render.go

```diff
@@ -399,17 +432,19 @@ func (r *Render) Data(w io.Writer, status int, v []byte) error {
 func (r *Render) HTML(w io.Writer, status int, name string, binding interface{}, htmlOpt ...HTMLOptions) error {
 
 	// If we are in development mode, recompile the templates on every HTML request.
-	if r.opt.IsDevelopment {
-		r.compileTemplates()
+	r.lock.RLock() // rlock here because we're reading the hasWatcher
```

Comment:
Do we need the lock to read `r.hasWatcher`? Looks like it's only set within the `CompileTemplates` func.

Reply:
@zeripath Looking again, the lock is probably a good idea since `CompileTemplates` is now exposed.

Reply:
Sorry I didn't get round to replying re: this - yes that was the reasoning.